filecoin-proofs/src/api/mod.rs

```diff
@@ -58,80 +53,40 @@ pub use winning_post::*;
 pub use storage_proofs_update::constants::partition_count;
 
 // Ensure that any associated cached data persisted is discarded.
-pub fn clear_cache<Tree: MerkleTreeTrait>(cache_dir: &Path) -> Result<()> {
+pub fn clear_cache(cache_dir: &Path) -> Result<()> {
     info!("clear_cache:start");
-
-    let mut t_aux: TemporaryAux<Tree, Sha256Hasher> = {
-        let f_aux_path = cache_dir.to_path_buf().join(CacheKey::TAux.to_string());
-        let aux_bytes = fs::read(&f_aux_path)
-            .with_context(|| format!("could not read from path={:?}", f_aux_path))?;
-
-        deserialize(&aux_bytes)
-    }?;
-
-    t_aux.set_cache_path(cache_dir);
-    let result = TemporaryAux::<Tree, DefaultPieceHasher>::clear_temp(t_aux);
-
+    let result = stacked::clear_cache_dir(cache_dir);
     info!("clear_cache:finish");
-
-    result
-}
-
-// Ensure that any associated cached data persisted is discarded.
-pub fn clear_caches<Tree: MerkleTreeTrait>(
-    replicas: &BTreeMap<SectorId, PrivateReplicaInfo<Tree>>,
-) -> Result<()> {
-    info!("clear_caches:start");
-
-    for replica in replicas.values() {
-        clear_cache::<Tree>(replica.cache_dir.as_path())?;
-    }
-
-    info!("clear_caches:finish");
-
-    Ok(())
-}
-
-// Ensure that any persisted layer data generated from porep are discarded.
-pub fn clear_layer_data<Tree: MerkleTreeTrait>(cache_dir: &Path) -> Result<()> {
-    info!("clear_layer_data:start");
-
-    let mut t_aux: TemporaryAux<Tree, Sha256Hasher> = {
-        let f_aux_path = cache_dir.to_path_buf().join(CacheKey::TAux.to_string());
-        let aux_bytes = fs::read(&f_aux_path)
-            .with_context(|| format!("could not read from path={:?}", f_aux_path))?;
-
-        deserialize(&aux_bytes)
-    }?;
-
-    t_aux.set_cache_path(cache_dir);
-    let result = TemporaryAux::<Tree, DefaultPieceHasher>::clear_layer_data(&t_aux);
-
-    info!("clear_layer_data:finish");
-
     result
 }
 
 // Ensure that any persisted vanilla proofs generated from synthetic porep are discarded.
-pub fn clear_synthetic_proofs<Tree: MerkleTreeTrait>(cache_dir: &Path) -> Result<()> {
+pub fn clear_synthetic_proofs(cache_dir: &Path) -> Result<()> {
     info!("clear_synthetic_proofs:start");
-
-    let mut t_aux: TemporaryAux<Tree, Sha256Hasher> = {
-        let f_aux_path = cache_dir.to_path_buf().join(CacheKey::TAux.to_string());
-        let aux_bytes = fs::read(&f_aux_path)
-            .with_context(|| format!("could not read from path={:?}", f_aux_path))?;
-
-        deserialize(&aux_bytes)
-    }?;
-
-    t_aux.set_cache_path(cache_dir);
-    let result = TemporaryAux::<Tree, DefaultPieceHasher>::clear_synthetic_proofs(&t_aux);
-
+    let result = stacked::clear_synthetic_proofs(cache_dir);
     info!("clear_synthetic_proofs:finish");
-
     result
 }
 
+/// Generate the replica id as expected for Stacked DRG.
+pub fn generate_replica_id<H: Hasher, T: AsRef<[u8]>>(
```

Comment:
Was this moved out of `storage-proofs-porep::stacked` just to reduce the API surface of that crate?

Reply:
I've moved that as the `stacked/vanilla/params.rs` looked like a "catch all" for all sorts of (unrelated) things. So I took the chance to move things around, while I was working on that file.